### PR TITLE
feat(controller): add runtime suggestion api

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

Add `/job/suggestion/runtime` API for getting the suggestion runtime list
- response the last used runtime if the model version is set (prefer using the online eval version)
- response the latest runtime in the project otherwise

demo 

```bash
curl -s 'https://example.com/api/v1/job/suggestion/runtime?projectId=1' | jq
```

```json
{
  "code": "success",
  "message": "Success",
  "data": {
    "runtimes": [
      {
        "id": "1",
        "name": "pn2olwnom7cml3pc545744fg7pfcpbsl6rb4h4i4",
        "tag": null,
        "alias": "v1",
        "meta": "artifacts:\n  dependencies:\n  - dependencies/requirements.txt\n  - dependencies/requirements-sw-lock.txt\n  files: []\n  runtime_yaml: runtime.yaml\n  wheels: []\nbase_image: ghcr.io/star-whale/starwhale:latest\nbuild:\n  os: Linux\n  sw_version: 0.0.0.dev0\nconfigs:\n  conda:\n    channels:\n    - conda-forge\n  docker:\n    image: ghcr.io/star-whale/runtime/pytorch\n  pip:\n    extra_index_url:\n    - https://mirrors.bfsu.edu.cn/pypi/web/simple\n    index_url: https://pypi.tuna.tsinghua.edu.cn/simple\n    trusted_host:\n    - pypi.tuna.tsinghua.edu.cn\n    - mirrors.bfsu.edu.cn\ncreated_at: 2023-02-07 16:53:45 CST\ndependencies:\n  conda_files: []\n  conda_pkgs: []\n  local_packaged_env: false\n  pip_files:\n  - requirements.txt\n  - requirements-sw-lock.txt\n  pip_pkgs:\n  - Pillow\n  - numpy\n  - scikit-learn\n  - torchvision\n  - torch\n  - torchdata\n  - torchtext\n  - torchaudio\n  - pycocotools\n  raw_deps:\n  - deps:\n    - Pillow\n    - numpy\n    - scikit-learn\n    - torchvision\n    - torch\n    - torchdata\n    - torchtext\n    - torchaudio\n    - pycocotools\n    kind: pip_pkg\n  - deps: requirements.txt\n    kind: pip_req_file\n  - deps: requirements-sw-lock.txt\n    kind: pip_req_file\nenvironment:\n  arch:\n  - noarch\n  auto_lock_dependencies: true\n  lock:\n    env_name: ''\n    env_prefix_path: ''\n    env_use_shell: false\n    shell:\n      python_env: conda\n      python_version: 3.7.16\n      use_conda: true\n      use_venv: false\n    starwhale_version: 0.3.6\n    system: Linux\n  mode: venv\n  python: '3.7'\nversion: pn2olwnom7cml3pc545744fg7pfcpbsl6rb4h4i4\n",
        "image": "ghcr.io/star-whale/starwhale:latest",
        "createdTime": 1675767600000,
        "owner": null
      }
    ]
  }
}
```

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Review by anda-ren — DISMISSED** [review]
LGTM

File: server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java [hunk 1]
```diff
@@ -130,5 +134,15 @@ public String updateSql(RuntimeVersionEntity version) {
                 }
             }.toString();
         }
+
+        public String findLatestByProjectIdSql(Long projectId, Integer limit) {
+            var ret = "select " + COLUMNS + " from runtime_version left join runtime_info on runtime_info.id ="
```
Comment (goldenxinxing) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java:139]: how about use SQL Builder? https://mybatis.org/mybatis-3/zh/statement-builders.html

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java]: Done

File: server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java [hunk 2]
```diff
@@ -130,5 +134,20 @@ public String updateSql(RuntimeVersionEntity version) {
...
         }
+
+        public String findLatestByProjectIdSql(Long projectId, Integer limit) {
+            return new SQL() {
+                {
+                    SELECT(COLUMNS);
+                    FROM("runtime_version");
+                    INNER_JOIN("runtime_info on runtime_info.id = runtime_version.runtime_id");
+                    WHERE("runtime_info.project_id = #{projectId}");
+                    ORDER_BY("runtime_version.modified_time desc");
```
[the remark this answers is not in the document]
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java:145]: Is it possible to use `runtime_version.id`?

[the remark this answers is not in the document]
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/mapper/RuntimeVersionMapper.java:145]: Added index to the runtime_version.modified_time
see https://github.com/star-whale/starwhale/pull/1805/commits/cd03a336fa92e794a2d46623aadf15526fc61f1d

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Review by dreamlandliu — APPROVED** [review]
LGTM

**Pull request merged**